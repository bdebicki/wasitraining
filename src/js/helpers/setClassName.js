export default function(element, elementClassNames) {
	const el = element;
	const classNames = elementClassNames;

	if (typeof (classNames) === 'string') {
		el.classList.add(classNames);
	} else if (Array.isArray(classNames)) {
		classNames.forEach((className) => {
			if (className) {
				el.classList.add(className);
			}
		});
	}
}
