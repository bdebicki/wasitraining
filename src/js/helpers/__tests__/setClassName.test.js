import using from '../../../../tests/utils/using';
import setClassName from '../setClassName';

const classNameCases = [
	{
		description: 'element with single class name',
		classNames: 'spanEl',
	},
	{
		description: 'element with multiple class name',
		classNames: [
			'spanEl',
			'spanEl--modifier',
		],
	},
];


describe('tests for set children helper function', () => {
	using(classNameCases).describe('render <span> element with specific class names', ({ description, classNames }) => {
		it(description, () => {
			// having
			const el = document.createElement('span');

			// when
			setClassName(el, classNames);

			// then
			expect(el).toMatchSnapshot();
		});
	});
});
