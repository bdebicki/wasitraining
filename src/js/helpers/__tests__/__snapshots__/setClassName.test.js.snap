// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for set children helper function render <span> element with specific class names element with multiple class name 1`] = `
<span
  class="spanEl spanEl--modifier"
/>
`;

exports[`tests for set children helper function render <span> element with specific class names element with single class name 1`] = `
<span
  class="spanEl"
/>
`;
